cfgloader/cfgloader.go

```diff
@@ -96,6 +105,10 @@ func Load(ctx context.Context, cfg any, opt ...Option) error {
 		}
 	}
 
+	if d, ok := cfg.(Defaultable); ok {
+		d.SetDefault()
```

Comment:
It's a little surprising to me that the defaults override the YAML. Have I understood correctly?

Reply:
That really depends on the `SetDefault()` logic. One should check zero values (ack the limitation on primitive types but it's not a major concern) before setting a default.

Reply:
SGTM, I'll go ahead and approve.

I'll continue making my feedback, which you can decide whether it makes sense to incorporate. If the SetDefault() call was first (before yaml), then the implementation of SetDefault would be impossible to do wrong. SetDefault could just blindly set the default values in each field, without any risk of the user forgetting or misunderstanding the need for nil checks.

Reply:
Appreciate the feedback!

I think your suggestion would work well except when the `SetDefault` is not self-contained, e.g. it needs to look at other config attributes and decide a default. E.g. In lumberjack we have a config to indicate where to fetch the ID token from the grpc metadata. We only want to set the default if the parent block is not nil.

Reply:
I ack that the current approach is not perfect. When we get the value from env var but we have `SetDefault` before loading the overwrites from env var. Maybe that's why I added `SetDefault` at last...

Reply:
Ah, I see now, thanks for explaining.